docs/rules/strict.md

```diff
@@ -139,6 +139,22 @@ function foo() {
 }());
 ```
 
+```js
+/*eslint strict: ["error", "function"]*/
+/*eslint-env es6*/
+
+// Illegal "use strict" directive in function with non-simple parameter list.
+// This is a syntax error since ES2016.
+function foo(a = 1) {
+    "use strict";
+}
+
+// We cannot write "use strict" directive in this function.
+// So we have to wrap this function with a function with "use strict" directive.
```

Comment:
The `strict` rule isn't currently in `"eslint:recommended"`, so users have to opt in to the rule. A mention about the "gotcha" in the opening text will give users a stronger incentive to opt in.